rasterio/rio/features.py

```diff
@@ -122,14 +122,26 @@ def mask(
         bounds = geojson.get('bbox', calculate_bounds(geojson))
 
         with rasterio.open(input) as src:
-            has_disjoint_bounds = disjoint_bounds(bounds, src.bounds)
+            # If y pixel value is positive, then invert y dimension in bounds
+            invert_y = src.affine.e > 0
+
+            src_bounds = src.bounds
+            if invert_y:
+                src_bounds = [src.bounds[0], src.bounds[3],
+                              src.bounds[2], src.bounds[1]]
+
```

Comment:
While working with NetCDF has turned my brain upside down, this case came from using a [test fixture raster](https://github.com/brendan-ward/rasterio/blob/overhaul_features_tests/tests/conftest.py#L192) with an `Affine.identity()` transform.

I too thought a refactor of the core to handle that better would be preferred.  This was one of those cases where the issue, once found, begged a pretty immediate fix where I detected it, rather than introduce the scope creep of a larger refactor.

We could either cut this out here and refactor that test fixture (possibly touching a bunch of tests along the way), or leave as a temporary shim and add an issue to address the problem in a more robust fashion.  Preference?